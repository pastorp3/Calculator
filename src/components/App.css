html {
  overflow-y: hidden;
}

.calculator {
  width: 700px;
}
