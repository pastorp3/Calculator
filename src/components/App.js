/* eslint-disable no-unused-vars, class-methods-use-this */
import './App.css';
import React from 'react';
import Display from './Display/Display';
import ButtonPanel from './ButtonPanel/ButtonPanel';
import calculate from '../logic/calculate';

class App extends React.Component {
  constructor() {
    super();

    this.state = {
      total: null,
      next: null,
      operator: null,
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(btnName) {
    const { total, next, operator } = this.state;
    console.log(total, next, operator);
    console.log(btnName);
  }

  render() {
    return (
      <div>
        <Display />
        <ButtonPanel clickHandler={() => this.handleClick} />
      </div>
    );
  }
}

export default App;

/* eslint-enable no-unused-vars, class-methods-use-this */
