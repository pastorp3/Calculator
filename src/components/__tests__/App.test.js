import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import App from '../App';

it('renders correctly', () => {
  const tree = renderer.create(<BrowserRouter><App /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});