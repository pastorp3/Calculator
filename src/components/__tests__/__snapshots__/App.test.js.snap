// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main>
  <div
    className="navbar"
  >
    <h1>
      Math Magicians
    </h1>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="mid-li"
      >
        <a
          href="/app"
          onClick={[Function]}
        >
          Claculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>
  <div
    className="container"
  >
    <h3>
      Let’s do some math!
    </h3>
    <div
      className="calculator"
    >
      <div
        className="display"
      >
        <span />
      </div>
      <div
        className="panel"
      >
        <div
          className="row"
        >
          <button
            className="btn-style btn"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
          <button
            className="btn-style btn"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            className="btn-style btn"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
          <button
            className="btn-style-def btn"
            onClick={[Function]}
            type="button"
          >
            ÷
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="btn-style btn"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            className="btn-style btn"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            className="btn-style btn"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
          <button
            className="btn-style-def btn"
            onClick={[Function]}
            type="button"
          >
            X
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="btn-style btn"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            className="btn-style btn"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            className="btn-style btn"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
          <button
            className="btn-style-def btn"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="btn-style btn"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            className="btn-style btn"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            className="btn-style btn"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
          <button
            className="btn-style-def btn"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="btn-style btn-0"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            className="btn-style btn"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
          <button
            className="btn-style-def btn"
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</main>
`;
